azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -255,17 +255,31 @@ private void updateExecutableFlow(final ExecutableFlow flow, final EncodingType
             + "SET status=?,update_time=?,start_time=?,end_time=?,enc_type=?,flow_data=? "
             + "WHERE exec_id=?";
 
-    final String json = JSONUtils.toJSON(flow.toObject());
     byte[] data = null;
     try {
+      // If this action fails, the execution must be failed.
+      final String json = JSONUtils.toJSON(flow.toObject());
       final byte[] stringData = json.getBytes("UTF-8");
       data = stringData;
       // Todo kunkun-tang: use a common method to transform stringData to data.
       if (encType == EncodingType.GZIP) {
         data = GZIPUtils.gzipBytes(stringData);
       }
     } catch (final IOException e) {
-      throw new ExecutorManagerException("Error encoding the execution flow.");
+      flow.setStatus(Status.FAILED);
+      updateExecutableFlowStatus(flow);
+      throw new ExecutorManagerException("Error encoding the execution flow. Execution Id  = "
+          + flow.getExecutionId());
+    } catch (final NullPointerException npe) {
```

Comment:
While using JSONUtils to read Object to string format it will throw RuntimeException and it is better to catch RuntimeException as it is parent of all the RuntimeException including NPE. And we could address the NPE here by adding the code mentioned in the earlier comment.

Reply:
Will do!